src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        int figureNumber = random.nextInt(5);
+        Figure[] figures = new Figure[]{
+                new IsoscelesTrapezoid(colorSupplier.getRandomColor(),random.nextInt(10),
+                         random.nextInt(10), random.nextInt(7)),
+                new Circle(colorSupplier.getRandomColor(), random.nextInt(10)),
+                new Rectangle(colorSupplier.getRandomColor(), random.nextInt(10),
+                        random.nextInt(10)),
+                new RightTriangle(colorSupplier.getRandomColor(), random.nextInt(10),
+                        random.nextInt(7)),
+                new Square(colorSupplier.getRandomColor(), random.nextInt(10))};
```

Comment:
Instead of passing expressions directly into the constructors, consider creating separate variables for each parameter. This will enhance code readability and align with the checklist recommendation: "Creating a figure, don't pass expressions in the constructor. Create separate variables and pass them on for better code readability."